src/com/ms/silverking/cloud/argus/Argus.java

```diff
@@ -7,180 +7,189 @@
-    private Terminator terminator;
-    private Terminator.KillType killtype;
-    private volatile ArgusTask    nextRSSTask;
-    private final RSSEnforcer    rssEnforcer;
-    private PeerWarningModule    peerWarningModule;
-    
-    public static long                lastPeerWarningMillis;
-    public static long                peerWarningResponseIntervalMillis;
-    
-    private static final int    defaultWarningFileIntervalSeconds = 60;
-    private static final long    defaultPeerWarningResponseIntervalSeconds = 10 * 60;
-    
-    public Argus(ArgusOptions options) throws IOException {
-        enforcers = new ArrayList<>();
-
-        killtype = Terminator.KillType.valueOf( options.terminatorType );
-        if (options.DiskUsageEnforcer != null) {
-            enforcers.add(initEnforcer(Test.DiskUsage, options.DiskUsageEnforcer, options));
-        }
-        enforcers.add(initEnforcer(Test.RSS, options.RSSEnforcer, options));
-        rssEnforcer = (RSSEnforcer)enforcers.get(enforcers.size() - 1);
-        
-        timer = new SafeTimer();
+  private static final String propKillEnabled = "killEnabled";
+  private static final String defaultKillEnabled = "false";
+  private static final String propEventsLogDir = "eventsLogDir";
+  private static final String defaultEventsLogDir = "/tmp";
+  private static final String udpPort = "udpPort";
+  private static final String warningFile = "warningFile";
+  private static final String warningFileIntervalSeconds = "warningFileIntervalSeconds";
+  private static final String peerWarningResponseIntervalSeconds = "peerWarningResponseIntervalSeconds";
+  private final List<SafetyEnforcer> enforcers;
+  private final SafeTimer timer;
+
+  private enum Test {RSS, DiskUsage}
+
+  ;
+  private boolean killEnabled;
+  private Terminator terminator;
+  private Terminator.KillType killtype;
+  private volatile SafeTimerTask nextRSSTask;
+  private final RSSEnforcer rssEnforcer;
+  private PeerWarningModule peerWarningModule;
+
+  public static long lastPeerWarningMillis;
+  public static long peerWarningResponseIntervalMillis;
+
+  private static final int defaultWarningFileIntervalSeconds = 60;
+  private static final long defaultPeerWarningResponseIntervalSeconds = 10 * 60;
+
+  public Argus(ArgusOptions options) throws IOException {
+    enforcers = new ArrayList<>();
+
+    killtype = Terminator.KillType.valueOf(options.terminatorType);
+    if (options.DiskUsageEnforcer != null) {
+      enforcers.add(initEnforcer(Test.DiskUsage, options.DiskUsageEnforcer, options));
+    }
+    enforcers.add(initEnforcer(Test.RSS, options.RSSEnforcer, options));
+    rssEnforcer = (RSSEnforcer) enforcers.get(enforcers.size() - 1);
+
+    timer = new SafeTimer();
+  }
+
+  private SafetyEnforcer initEnforcer(Test test, String testSpec, ArgusOptions options) throws IOException {
+    Properties prop = new Properties();
+    try {
+      FileInputStream fis = new FileInputStream(testSpec);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code contains a resource that might not be closed properly. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the following resource: *fis*. The resource is referenced in statements at the following line: 71. The resource closure statement is at line: 72. There are other execution paths that do not contain closure statements, for example, when Properties.load throws an exception. Either a) close *fis* in a try-finally block or b) close the resource by declaring *fis* in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).